module/trace/noop.go

```diff
@@ -73,19 +76,21 @@ func (t *NoopTracer) StartTransactionSpan(
 	return NoopSpan, ctx
 }
 
-func (t *NoopTracer) StartSpanFromContext(
-	ctx context.Context,
+func (t *NoopTracer) StartSpanFromParent(
+	parentSpan trace.Span,
 	operationName SpanName,
 	opts ...trace.SpanStartOption,
-) (
-	trace.Span,
-	context.Context,
-) {
-	return NoopSpan, ctx
+) trace.Span {
+	return NoopSpan
 }
 
-func (t *NoopTracer) StartSpanFromParent(
+func (t *NoopTracer) ShouldSample(entityID flow.Identifier) bool {
+	return true
```

Comment:
should it return `false` instead, since we don't want to trace any.

Reply:
it doesn't matter since the child span would be a no-op span either way